Lib/Protocols/IdFTPServer.pas

```diff
@@ -6018,6 +6053,56 @@ function TIdFTPServer.MLSFEATLine(const AFactMask: TIdMLSDAttrs;
   end;
 end;
 
+function ParseCSIDParams(const AParams : String) : string;
+var LBuf,
+    LValueValue : String;
+    LValueName : String;
+    LCltName : String;
+    LCltVersion : String;
+begin
+   Result := '';
+   LBuf := AParams;
+   repeat
+      LValueValue := Fetch(LBuf,';');
+      LValueName := TrimLeft(Fetch(LValueValue,'='));
+      case PosInStrArray(LValueName,['Name','Version']) of
+        0 : LCltName := LValueValue;
+        1 : LCltVersion := LValueValue;
+      end;
+   until LBuf = '';
+   Result := Trim(LCltName + ' ' + LCltVersion);
+end;
+
+procedure TIdFTPServer.CommandCSID(ASender: TIdCommand);

```

Comment:
A CSID response can contain additional name=value pairs (os-dependant-fact, local-fact, etc) so I would suggest adding a way for the user to provide those values as well.